src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs

```diff
@@ -1016,8 +1016,8 @@ public int GatherFields(ReadOnlyMemory<char> lineSpan, ReadOnlySpan<char> span,
                                 }
                                 var spanT = Fields.Spans[Fields.Count - 1];
 
-                                int csrc = 0;
-                                if (!Conversions.Instance.TryConvert(in spanT, ref csrc) || csrc <= 0)
+                                int csrc;
+                                if (!Conversions.Instance.TryParse(in spanT, out csrc) || csrc <= 0)
```

Comment:
Note that TryParse is going to be slightly more expensive, but it probably doesn't matter.

Reply:
> slightly more expensive

Because of the extra write to the `out` parameter? Or because of the under/overflow check? Or something else?

Reply:
The extra write in the case of success, two extra writes in the case of failure, the overflow check, the cast from the nullable rather than GetValueOrDefault, larger code that's even less likely to be inlined (though it's probably already not getting inlined).  In general this code is not particularly clean.